app/controllers/MakeFileConfidentialController.scala

```diff
@@ -54,7 +54,18 @@ class MakeFileConfidentialController @Inject() (
     request: DataRequest[?]
   ): HtmlFormat.Appendable = {
     val fileId: String = request.userAnswers.get(UploadSupportingMaterialMultiplePage).map(_.last.id).get
-    makeFileConfidentialView(appConfig, preparedForm, submitAction, mode, fileId)
+      makeFileConfidentialView(appConfig, preparedForm, submitAction, mode, fileId)
+  }
+
+  override def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request:DataRequest[?] =>
+      val files = request.userAnswers.get(UploadSupportingMaterialMultiplePage)
```

Comment:
if files is None, calling files.get will throw a NoSuchElementException at runtime, causing a server error. Using the followng pattern matching should guard against this too

      request.userAnswers.get(UploadSupportingMaterialMultiplePage) match {
        case Some(files) if files.nonEmpty => Ok(renderView(form, submitAction(mode), mode))
        case _ => Redirect(routes.UploadSupportingMaterialMultipleController.onPageLoad(Some(request.id.toString), mode))
      }